### PR TITLE
Create plot and export model

### PR DESCRIPTION
**Pull request by aflag**
There are a few points to be made in this PR:

1. Since I already created a toPandas function to make plotting easier, I took the opportunity to also output the model into a csv file (we can easily modify the format later, I didn't want to start off with hdf5 because that requires an extra dependency)
2. I needed a way to match plots and other such data, so I've created a very simple library in tests.utils. Please take a look. I can use it in the other regression tests if we all like it.
3. I've implemented model averaging here: https://github.com/ScottishCovidResponse/simple_network_sim/pull/43/files#diff-678f9ead2fe6a969469bd9db5f08c916R57-R66. I think I could make that as part of a library function rather than in the main function, but I just wanted @magicicada's input if that's even useful

https://github.com/ScottishCovidResponse/SCRCIssueTracking/issues/216
https://github.com/ScottishCovidResponse/SCRCIssueTracking/issues/324

### CODE REVIEW

**Comment by aflag** [comment]
This is how the reult of running the plot function: test

File: tests/unit/test_network_of_populations.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 import networkx as nx
 import pytest
 
-from simple_network_sim import network_of_populations as np, loaders
```
Comment (bobturneruk) [inline tests/unit/test_network_of_populations.py:9]: What has prompted this change, please?

File: simple_network_sim/network_of_populations.py [hunk 2]
```diff
@@ -300,6 +301,76 @@ def exposeRegions(infections, states):
...
+    Takes a network of population timeseries (dictOfStates and transforms it into a pandas DataFrame.
+
+    :param timeseries: dict object with the state over the times
+    :return: a pandas dataframe in tabular format with the following columns:
+             - time
+             - healthboard
+             - age
+             - state
+             - total
+    """
```
Comment (bobturneruk) [inline simple_network_sim/network_of_populations.py:315]: Love the docstrings!

**Comment by aflag** [comment]
> Is this code called anywhere at present? Perhaps I'm missing something?

no, it's not finished, that's why I raised it as a draft. I just wanted early feedback for the direction I'm taking.

File: tests/unit/test_network_of_populations.py (hunk 1 above)
Comment (aflag) [reply tests/unit/test_network_of_populations.py:9]: I think the line below this is `from simple_network_sim import network_of_populations as np`. `loaders` was not used in the file, so I removed it (it was my IDE's idea)

**Comment by bobturneruk** [comment]
@aflag - my bad!

**Comment by bobturneruk** [comment]
Could the `::`s be removed from file names, please? Windows won't let me pull as the characters are not allowed in paths! Happy to switch to Linux if this is problematic.

**Comment by bobturneruk** [comment]
I think it would be useful to include `sample_output_files` with a `.pdf` and `.csv` so we can illustrate to other teams the kind of things we're generating without them having to run the code.

**Comment by aflag** [comment]
@bobturneruk I've added the sample output. The csv file can be easily read into a pandas dataframe with ```pandas.read_csv(<filename>)```.

**Comment by magicicada** [comment]
The averaging is going to be useful, thanks for including it.

File: tests/utils.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    This function is specialised in comparing matplotlib figures with a baseline. The baseline will be an image inside
+    mpl_images in the current directory, with name matching the test's name.
+
+    The baseline will be automatically created if it doesn't exist (but the test will still fail). Using force_update
+    allows the user to iterate over how the image looks through running the tests multiple times with the flag enabled.
+
+    :param fig: The produced image you want to check for regressions
+    :param force_update: This can be handy during development. Setting this to True will always fail the test, but the
+                         baseline image will be updated, so you can easily iterate over the way you want the plotting
+                         function called
```
Comment (bobturneruk) [inline tests/utils.py:20]: I like this approach.

**Comment by bobturneruk** [comment]
Resolves https://github.com/ScottishCovidResponse/SCRCIssueTracking/issues/323

**Comment by bobturneruk** [comment]
Is there a stray comma at the start of the `.csv` output?

**Comment by bobturneruk** [comment]
Maybe it's a row number? Probably shouldn't be there...

**Comment by aflag** [comment]
@bobturneruk it actually has to be there. That's just the output of pd.DataFrame.to_csv. The reason it's there is because that's the index column, which doesn't have a name. I could call `to_csv(index=False)` (that column is essentially just the line number right now). I will do that, it's redundant information that we don't need.

**Comment by bobturneruk** [comment]
@aflag I think that's wise.

**Pull request merged**